### ISSUE TITLE
复现不出论文的结果

### ISSUE THREAD

**Issue by Git-Shaw** [CCIIPLab/BiSyn_GAT_plus#6]
您好，感谢开源代码。

我参考了Issues中您提供的超参数，但分数仍旧差距3~4个百分点，以下是我尝试的超参：
Loading vocab from: data/V2/Laptops/vocab_tok.vocab
Loading vocab from: data/V2/Laptops/vocab_pol.vocab
token_vocab: 3525, polarity_vocab:3
-----------  Configuration Arguments -----------
aa_graph_self: True
aa_graph_version: 1
aa_num_layer: 6
adj_span_version: 0
aspect_graph_encoder_version: 1
aspect_graph_num_layer: 1
attn_dropout: 0.0
attn_head: 2
average_mapback: True
batch_size: 32
bert_hidden_dim: 768
bert_lr: 2e-05
bidirect: True
borrow_encoder: False
con_dep_conditional: True
con_dep_version: con_dot_dep
data_dir: data/V2/Laptops
device: cuda
hidden_dim: 200
input_dropout: 0.1
is_average: True
is_filtered: True
l2: 1e-05
layer_dropout: 0.2
leaf2root: False
log_step: 16
lower: True
lr: 3e-05
max_len: 100
max_num_spans: 3
max_patience: 20
need_preprocess: False
num_class: 3
num_encoder_layer: 3
num_epoch: 20
plus_AA: True
rnn_hidden_dim: 200
rnn_layers: 1
root2leaf: False
seed: 2
sort_key_idx: 0
special_token: [N]
split_aa_graph: True
tok_size: 3525
vocab_dir: data/V2/Laptops

In Results: test_epoch:7, test_acc:78.79746835443038, test_f1:0.7479312410299718
Last In Results: test_epoch:20, test_acc:76.42405063291139, test_f1:0.714276633613018

第二套超参
Loading vocab from: data/V2/Laptops/vocab_tok.vocab
Loading vocab from: data/V2/Laptops/vocab_pol.vocab
token_vocab: 3525, polarity_vocab:3
-----------  Configuration Arguments -----------
aa_graph_self: True
aa_graph_version: 1
aa_num_layer: 6
adj_span_version: 0
aspect_graph_encoder_version: 1
aspect_graph_num_layer: 2
attn_dropout: 0.1
attn_head: 2
average_mapback: True
batch_size: 16
bert_hidden_dim: 768
bert_lr: 2e-05
bidirect: True
borrow_encoder: False
con_dep_conditional: True
con_dep_version: con_dot_dep
data_dir: data/V2/Laptops
device: cuda
hidden_dim: 200
input_dropout: 0.1
is_average: True
is_filtered: True
l2: 1e-05
layer_dropout: 0.1
leaf2root: False
log_step: 16
lower: True
lr: 1e-05
max_len: 90
max_num_spans: 3
max_patience: 20
need_preprocess: False
num_class: 3
num_encoder_layer: 3
num_epoch: 20
plus_AA: True
rnn_hidden_dim: 200
rnn_layers: 1
root2leaf: False
seed: 9
sort_key_idx: 0
special_token: [N]
split_aa_graph: True
tok_size: 3525
vocab_dir: data/V2/Laptops

请问我的设置有什么问题吗？或者您能提供一个完整的超参数吗？

**Comment by dreamnlp** [comment]
> 您好，感谢开源代码。
> 
> 我参考了Issues中您提供的超参数，但分数仍旧差距3~4个百分点，以下是我尝试的超参： Loading vocab from: data/V2/Laptops/vocab_tok.vocab Loading vocab from: data/V2/Laptops/vocab_pol.vocab token_vocab: 3525, polarity_vocab:3 ----------- Configuration Arguments ----------- aa_graph_self: True aa_graph_version: 1 aa_num_layer: 6 adj_span_version: 0 aspect_graph_encoder_version: 1 aspect_graph_num_layer: 1 attn_dropout: 0.0 attn_head: 2 average_mapback: True batch_size: 32 bert_hidden_dim: 768 bert_lr: 2e-05 bidirect: True borrow_encoder: False con_dep_conditional: True con_dep_version: con_dot_dep data_dir: data/V2/Laptops device: cuda hidden_dim: 200 input_dropout: 0.1 is_average: True is_filtered: True l2: 1e-05 layer_dropout: 0.2 leaf2root: False log_step: 16 lower: True lr: 3e-05 max_len: 100 max_num_spans: 3 max_patience: 20 need_preprocess: False num_class: 3 num_encoder_layer: 3 num_epoch: 20 plus_AA: True rnn_hidden_dim: 200 rnn_layers: 1 root2leaf: False seed: 2 sort_key_idx: 0 special_token: [N] split_aa_graph: True tok_size: 3525 vocab_dir: data/V2/Laptops
> 
> In Results: test_epoch:7, test_acc:78.79746835443038, test_f1:0.7479312410299718 Last In Results: test_epoch:20, test_acc:76.42405063291139, test_f1:0.714276633613018
> 
> 第二套超参 Loading vocab from: data/V2/Laptops/vocab_tok.vocab Loading vocab from: data/V2/Laptops/vocab_pol.vocab token_vocab: 3525, polarity_vocab:3 ----------- Configuration Arguments ----------- aa_graph_self: True aa_graph_version: 1 aa_num_layer: 6 adj_span_version: 0 aspect_graph_encoder_version: 1 aspect_graph_num_layer: 2 attn_dropout: 0.1 attn_head: 2 average_mapback: True batch_size: 16 bert_hidden_dim: 768 bert_lr: 2e-05 bidirect: True borrow_encoder: False con_dep_conditional: True con_dep_version: con_dot_dep data_dir: data/V2/Laptops device: cuda hidden_dim: 200 input_dropout: 0.1 is_average: True is_filtered: True l2: 1e-05 layer_dropout: 0.1 leaf2root: False log_step: 16 lower: True lr: 1e-05 max_len: 90 max_num_spans: 3 max_patience: 20 need_preprocess: False num_class: 3 num_encoder_layer: 3 num_epoch: 20 plus_AA: True rnn_hidden_dim: 200 rnn_layers: 1 root2leaf: False seed: 9 sort_key_idx: 0 special_token: [N] split_aa_graph: True tok_size: 3525 vocab_dir: data/V2/Laptops
> 
> 请问我的设置有什么问题吗？或者您能提供一个完整的超参数吗？

同求，我是差2-3个百分点

**Comment by YouAreAllINeed** [comment]
我也是，我直接运行train.py，得到的结果都和论文的里的差很多。

**Comment by dreamnlp** [comment]
我参考了Issues中您提供的超参数

为啥我laptop数据集batch_size设置成16，training set只有91，而您的日志是143？

**Comment by ShuoLiang-98** [comment]
您好，https://github.com/CCIIPLab/BiSyn_GAT_plus/issues/5#issuecomment-1518923696 提供的log文件里面应该打印出了完整的超参列表，您可以参考设置dropout、learning_rate、batch_size等重要参数。（概述：先将plus_AA设置为False训练BiSyn-GAT，再将其设置为True并加载之前训练好的BiSyn-GAT模型，训练BiSyn-GAT+）
